pkg/segment/writer/metrics/metricssegment.go

```diff
@@ -522,6 +527,38 @@ func ExtractOTSDBPayload(rawJson []byte, tags *TagsHolder) ([]byte, float64, uin
 						ts = uint32(intVal)
 					}
 				}
+			case jp.String:
+				// First, try to parse the date as a number (seconds or milliseconds since epoch)
+				if t, err := strconv.ParseInt(string(value), 10, 64); err == nil {
+					// Determine if the number is in seconds or milliseconds
+					if toputils.IsTimeInMilli(uint64(t)) {
+						ts = uint32(t / 1000)
+					} else {
+						ts = uint32(t)
+					}
+				}
+
+				// Parse the string to time using time.Parse and multiple layouts.
+				layouts := []string{
```

Comment:
If you don't get an error on line 532, you can break out of this early before continuing with parsing it according to various RFCs, right?

Reply:
Ahh, yes, I added a return inside the `If` block.